addons/promote/common/src/main/java/org/commonjava/indy/promote/data/PromotionManager.java

```diff
@@ -761,430 +561,324 @@ public PathsPromoteResult rollbackPathsPromote( final PathsPromoteResult result
         {
-            ArtifactStore src = source;
-            byPathTargetLocks.lockAnd( targetKey, config.getLockTimeoutSeconds(), k -> {
-                for ( final Transfer transfer : contents )
+            return conflictManager.checkAnd( k, pathsLockKey -> {
+                ValidationResult validationResult = new ValidationResult();
+                if ( !skipValidation )
                 {
-                    if ( transfer != null && transfer.exists() )
+                    try
                     {
-                        try
-                        {
-                            if ( copyToSource )
-                            {
-                                final String path = transfer.getPath();
-                                try (InputStream stream = transfer.openInputStream( true ))
-                                {
-                                    contentManager.store( src, path, stream, TransferOperation.UPLOAD,
-                                                          new EventMetadata() );
-                                }
-                                catch ( IndyWorkflowException e )
-                                {
-                                    wfEx.set( e );
-                                    return null;
-                                }
-                            }
+                        ValidationRequest vr = validator.validate( request, validationResult, baseUrl );
+                    }
+                    catch ( Exception e )
+                    {
+                        ex.set( e );
+                        return null;
+                    }
+                }
 
-                            transfer.delete( true );
-                            completed.remove( transfer.getPath() );
-                            pending.add( transfer.getPath() );
-                        }
-                        catch ( final IOException e )
-                        {
-                            String msg = String.format( "Failed to rollback path promotion of: %s from: %s. Reason: %s",
-                                                        transfer, result.getRequest().getSource(), e.getMessage() );
-                            logger.error( msg, e );
-                            error.set( msg );
-                        }
+                if ( validationResult.isValid() )
+                {
+                    if ( request.isDryRun() )
+                    {
+                        return new PathsPromoteResult( request, pending, emptySet(), emptySet(), validationResult );
+                    }
+
+                    PathsPromoteResult result = runPathPromotions( request, pending, contents, validationResult );
+                    if ( result.succeeded() )
+                    {
+                        promotionHelper.updatePathPromoteMetrics( contents.size(), result );
                     }
+
+                    return result;
```

Comment:
We might want to add an else clause here and log the validation errors, so we can help with debugging.